src/common/roll_renderer.js

```diff
@@ -610,6 +609,52 @@ class Beyond20RollRenderer {
                         break;
                     }
                 }
+            } else if (request.name.includes("Sorcerous Burst")) {
+                const mods = Object.keys(request.character.spell_modifiers);
+                let burstRollsLeft = 0;
+                let burstRollCount = 0;
+                let selectedModifier = null;
+                for (let [i, dmg_roll] of damage_rolls.entries()) {
+                    const [dmg_type, roll, flags] = dmg_roll;
+                    const faces = parseInt(roll.dice[0].faces);
+                    const burstCount = roll.dice[0].rolls.filter(x => x.roll === faces).length; // Count how many max rolls occurred
+
+                    if (burstCount > 0) {
+                        burstRollCount++;
+                        // Prompt for the ability modifier if needed (only if there are multiple options)
+                        if (mods.length > 1 && selectedModifier === null) {
+                            const result = await this.queryGeneric("Spellcasting Ability Modifier", "Burst Ability Modifier", mods);
+                            selectedModifier = (result !== null) ? mods[result] : mods[0]; // Use first one if not selected.
+                        } else { 
+                            selectedModifier = mods[0];
+                        }
+
+                        const modifier = parseInt(request.character.spell_modifiers[selectedModifier]);
+                        burstRollsLeft = (burstRollsLeft != 0) ? burstRollsLeft : modifier; // Set burstRollsLeft to the modifier value if not already set
+                        
+                        // Calculate how many rolls are allowed for the burst (delta)
+                        const delta = Math.min(burstCount, burstRollsLeft); // Only allow as many burst rolls as available in burstRollsLeft
+                        
+                        // Reduce burstRollsLeft by delta, ensuring it doesn't go below zero
+                        burstRollsLeft -= delta;
+
+                        // Perform the burst rolls if delta > 0
+                        if (delta > 0 && burstRollsLeft >= 0) {
```

Comment:
This should only check for `delta`, otherwise if burstRollsLeft decreases to 0 after you substract delta from it on the previous line, it will make you skip this block by mistake.